containers/nestjs/src/game/APong.ts

```diff
@@ -71,8 +83,15 @@ class Velocity {
   set len(value: number) {
     {
       const fact = value / this.len;
-      this._dx *= fact;
-      this._dy *= fact;
+      // The if statement is put here to prevent the ball from gaining too much speed and going through the player paddle
+      if (this._dx * fact <= 25) {
+        this._dx *= fact;
+        this._dy *= fact;
+      } else {
+        const num: number = 25 / this._dx;
+        this._dx *= num;
+        this._dy;
```

Comment:
The variable name `num` could be made more descriptive, and the explicit `: number` is unnecessary since it is implied from the `25` that is being assigned.

The `this._dy;` line seems a bit odd, cause it doesn't do anything, making me wonder if you intended to do something there or just forgot to remove this line.

I can't really wrap my head around `25 / this._dx`. I think the result'd get bigger the smaller the old dx is, but then the new dx is the old dx times the result, so I think they cancel out to just always assigning the value 25?:
`this._dx *= 25 / this._dx`
`this._dx = this._dx * (25 / this._dx)`
`this._dx = 25`

If you want to limit dx to 25, I recommend getting rid of the if-statement and using `Math.min(n, 25)` to clamp the value.

Reply:
Mmh, weird, line 93 should have been `this._dy *= num;`, but it seems like the last part got left out. I don't know how, as I never deleted that during testing but I'll place it back!

This was made originally so `this._dx` never exceeds 25 while not fucking up the ratio between _dx and _dy.

I will also look into a better name for `num`